### PR TITLE
Initial commit

### LINKED ISSUE #3: Initial commit
Get the initial project structure in place, with the basic code & units working too.

### PR DESCRIPTION
**Pull request by grant-d**
@reisenberger please would you take a look at this.
- The initial commit for this repo, which I created by using the provided template.
- I am not familiar with your build system, versioning & other conventions so all comments welcome.
- If it's easier, feel free to push directly to this PR, else I can merge then you can edit in a separate PR.
- If you can please ensure it's using a **preview version no** so that we don't publish a release Nuget yet

- Would you mind setting up CI & any policies you use on the main repo (#4 )

Resolves #3

### CODE REVIEW

**Comment by reisenberger** [comment]
@grant-d Many thanks for this!  I will get to this for review as soon as I can.

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,6 @@ TestResults
 *.sln.docstates
 .vs/
 .vscode/
-.idea/
```
Comment (reisenberger) [inline .gitignore]: This line can be worth re-instated, as helps Mac Rider users

File: .gitignore [hunk 2]
```diff
@@ -105,8 +104,6 @@ ClientBin
 [Ss]tyle[Cc]op.*
 ~$*
 *.dbmdl
-.DS_Store
```
Comment (reisenberger) [inline .gitignore]: Think a Mac user added this `.DS_Store` to help Rider users - can re-instate

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Polly.Contrib.BlankTemplate changelog
```
Comment (reisenberger) [inline CHANGELOG.md]: Polly.Contrib.WaitAndRetry

File: src/Polly.Contrib.WaitAndRetry.Specs/ExponentialBackoffSpecs.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            // Act
+            IEnumerable<TimeSpan> result = Backoff.ExponentialBackoff(initialDelay, retryCount, factor, fastFirst);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void Backoff_WithFactorLessThanZero_ThrowsException()
```
Comment (reisenberger) [inline src/Polly.Contrib.WaitAndRetry.Specs/ExponentialBackoffSpecs.cs]: LessThanOne

File: src/Polly.Contrib.WaitAndRetry.Specs/ExponentialBackoffSpecs.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                }
+                else
+                {
+                    timeSpan.Should().BeGreaterOrEqualTo(initialDelay);
+                }
+            }
+        }
+
+        [Fact]
+        public void Backoff_ResultIsInRange()
```
Comment (reisenberger) [inline src/Polly.Contrib.WaitAndRetry.Specs/ExponentialBackoffSpecs.cs]: We haven't strictly got a test that `factor` is correctly applied.  Could have a test similar to this one with an assertion:

    result.Select(t => t.TotalMilliseconds).Should().BeEquivalentTo(new double[] {10, 20, 40, 80, 160});

(or similar)

File: src/Polly.Contrib.WaitAndRetry.Specs/LinearBackoffSpecs.cs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            result.Should().HaveCount(retryCount);
+
+            foreach (TimeSpan timeSpan in result)
+            {
+                timeSpan.Should().Be(initialDelay);
+            }
+        }
+
+        [Fact]
+        public void Backoff_ResultIsIncrementing()
```
Comment (reisenberger) [inline src/Polly.Contrib.WaitAndRetry.Specs/LinearBackoffSpecs.cs]: We haven't strictly got a test that `factor` is applied properly.  Could have a test similar to this with `factor = 2` and a simple:

            result.Select(t => t.TotalMilliseconds).Should().BeEquivalentTo(new double[] { 10, 30, 50, 70, 90 });

(or similar)

File: src/Polly.Contrib.WaitAndRetry/Backoff.DecorrelatedJitter.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        /// For example: 117ms, 236ms, 141ms, 424ms, ...
+        /// For background, see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/.
+        /// </summary>
+        /// <param name="minDelay">The minimum duration value to use for the wait before each retry.</param>
+        /// <param name="maxDelay">The maximum duration value to use for the wait before each retry.</param>
+        /// <param name="retryCount">The maximum number of retries to use, in addition to the original call.</param>
+        /// <param name="fastFirst">Whether the first retry will be immediate or not.</param>
+        /// <param name="seed">An optional <see cref="Random"/> seed to use.
+        /// If not specified, will use a shared instance with a random seed, per Microsoft recommendation for maximum randomness.</param>
+        public static IEnumerable<TimeSpan> DecorrelatedJitterBackoff(TimeSpan minDelay, TimeSpan maxDelay, int retryCount, bool fastFirst = false, int? seed = null)
```
Comment (reisenberger) [inline src/Polly.Contrib.WaitAndRetry/Backoff.DecorrelatedJitter.cs]: Where other examples had more than one optional parameter, `bool fastFirst = false` always appeared last.  Should we make it consistent?  (No opinion which way round it should be fastFirst last or earlier)

**Comment by reisenberger** [comment]
Huge thanks for your time on this @grant-d . Looking lovely to me. :+1:  (Commented some tiny things - apologies for separate email comments - meant to `Start a review` but kept hitting `Single comment`)

**Comment by grant-d** [comment]
No problem, CR complete

**Pull request merged**